lib/ReactViews/Map/Panels/SharePanel/BuildShareLink.ts

```diff
@@ -255,8 +256,12 @@ export function isShareable(terria: Terria) {
     const model = terria.getModelById(BaseModel, modelId);
     return (
       model &&
-      ((HasLocalData.is(model) && !model.hasLocalData) ||
-        !HasLocalData.is(model))
+      // If this is a Reference, and has .target defined, perform checks against model.target
+      (ReferenceMixin.isMixedInto(model) && model.target
+        ? (HasLocalData.is(model.target) && !model.target.hasLocalData) ||
+          !HasLocalData.is(model.target)
+        : (HasLocalData.is(model) && !model.hasLocalData) ||
+          !HasLocalData.is(model))
```

Comment:
The reference check can be replaced with `getDereferencedIfExists`

Reply:
Great that we have this, handy. Implemented but not the most elegant. Not sure most consise way to write but this is correct I think! Open to suggestions.